Spigot-Server-Patches/0619-Implement-a-method-to-get-tile-entities-in-a-chunk-b.patch

```diff
@@ -0,0 +1,50 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Matthew Miller <mnmiller1@me.com>
+Date: Fri, 18 Dec 2020 21:05:55 +1000
+Subject: [PATCH] Implement a method to get tile entities in a chunk by
+ Predicate
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftChunk.java b/src/main/java/org/bukkit/craftbukkit/CraftChunk.java
+index 42b47634437135a9e9b608283f3ce81c98ca181a..284e78c8171a9e50a4267433a953f4d9e8580f37 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftChunk.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftChunk.java
+@@ -3,8 +3,10 @@ package org.bukkit.craftbukkit;
+ import com.google.common.base.Preconditions;
+ import com.google.common.base.Predicates;
+ import java.lang.ref.WeakReference;
++import java.util.ArrayList;
+ import java.util.Arrays;
+ import java.util.Collection;
++import java.util.List;
+ import java.util.function.Predicate;
+ import net.minecraft.server.BiomeStorage;
+ import net.minecraft.server.BlockPosition;
+@@ -157,6 +159,27 @@ public class CraftChunk implements Chunk {
+         return entities;
+     }
+ 
++    // Paper start
++    @Override
++    public Collection<BlockState> getTileEntities(Predicate<Block> blockPredicate, boolean useSnapshot) {
++        if (!isLoaded()) {
++            getWorld().getChunkAt(x, z); // Transient load for this tick
++        }
++        net.minecraft.server.Chunk chunk = getHandle();
++
++        List<BlockState> entities = new ArrayList<>();
```

Comment:
java.util.List<BlockState> entities = new java.util.ArrayList<>();

Reply:
Please refrain from leaving reviews pointing out imports issues in the future. They are incredibly minor and unhelpful, and any imports that we feel shouldn't make it in the patch we can easily handle during merging.